### DISCUSSION TITLE
i have runtime error with linear regression function

### DISCUSSION THREAD

**Discussion by Abdusattor20030610** [mrdbourke/pytorch-deep-learning#300] [Q&A] [answered]
in first linear regression function there is error in loss=loss_fn(y_pred,y_train) |error says The size of tensor a (10) must match the size of tensor b (40) at non-singleton dimension 0

**Comment by nazarPuriy** [comment, marked as answer]
That means that the dimension is not the same, you have 2 arrays with different length. Probably you are predicting on test and then trying to match it with the train or something like that.

**Reply by mrdbourke** [reply]
Yes, @nazarPuriy is correct. It means there's a dimension/shape issue with tensors.

Try printing out the shapes of your tensors and see if you can line them up for the equation you're trying to perform.

See this page for more details on shape errors: https://www.learnpytorch.io/pytorch_most_common_errors/#1-shape-errors-in-pytorch
